opm/autodiff/BlackoilPressureModel.hpp

```diff
@@ -49,8 +49,35 @@ namespace Opm {
         typedef typename Base::SolutionState SolutionState;
         typedef typename Base::V V;
 
-        // Inherit constructor.
-        using Base::Base;
+        /// Construct the model. It will retain references to the
+        /// arguments of this functions, and they are expected to
+        /// remain in scope for the lifetime of the solver.
+        /// \param[in] param            parameters
+        /// \param[in] grid             grid data structure
+        /// \param[in] fluid            fluid properties
+        /// \param[in] geo              rock properties
+        /// \param[in] rock_comp_props  if non-null, rock compressibility properties
+        /// \param[in] wells_arg        well structure
+        /// \param[in] linsolver        linear solver
+        /// \param[in] eclState         eclipse state
+        /// \param[in] has_disgas       turn on dissolved gas
+        /// \param[in] has_vapoil       turn on vaporized oil feature
+        /// \param[in] terminal_output  request output to cout/cerr
+        BlackoilPressureModel(const typename Base::ModelParameters&   param,
+                              const Grid&                             grid,
+                              const BlackoilPropsAdInterface&         fluid,
+                              const DerivedGeology&                   geo,
+                              const RockCompressibility*              rock_comp_props,
+                              const StandardWells&                    std_wells,
+                              const NewtonIterationBlackoilInterface& linsolver,
+                              Opm::EclipseStateConstPtr               eclState,
```

Comment:
> the parser module is seeing a lot of `shared_ptr<>` removal these days.

Sure, but that's sort of tangential.  How we construct interfaces that consume `EclipseState` information should be decoupled from the representation within the parser.  While the parser exposes `shared_ptr<>`s I'd prefer a single dereference operation at the top of the call tree and that lower levels be implemented in terms of reference-to-const.